File: leetcode/s0001_0100/leetcode_1.py
from typing import *
class leetcode_1(object):
    """leetcode_1 documentation"""

    def __init__(self, *args, **kwargs):
        """Constructor for leetcode_1"""
        pass

    def solve(self):
        print(self.twoSum([2, 7, 11, 15], 9))

    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_dict = dict()

        for id, num in enumerate(nums):
            if str(target - num) not in num_dict:
                num_dict.update(**{
                    str(num): id
                })
            else:
                return [num_dict[str(target-num)], id]

        return [-1, -1]


if __name__ == '__main__':
    l: leetcode_1 = leetcode_1()
    l.solve()

File: src/easy/leetcode_1.java
package easy;

import java.util.HashMap;

public class leetcode_1 {
    public void solve() {
    }

    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> a = new HashMap<>();
        for(int i=0;i<nums.length;i++) {
            int temp = target-nums[i];

            if(a.containsKey(temp)) {
                return new int[]{a.get(temp), i};
            } else {
                a.put(nums[i], i);
            }
        }
        return null;
    }

    public static void main(String[] args) {
        leetcode_1 t = new leetcode_1();
        t.solve();
    }
}
